HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -2178,6 +2180,45 @@ def self.dehumidifier_ief_to_ef_inputs(w_coeff, ef_coeff, ief, water_removal_rat
     return ef_input, water_removal_rate_input
   end
 
+  def self.adjust_dehumidifier_load_EMS(fraction_served, zone_hvac, model, living_space)
+    # sensor
+    dehumidifier_sens_htg = OpenStudio::Model::EnergyManagementSystemSensor.new(model, 'Zone Dehumidifier Sensible Heating Rate')
+    dehumidifier_sens_htg.setName("#{zone_hvac.name} sens htg")
+    dehumidifier_sens_htg.setKeyName(zone_hvac.name.to_s)
+    dehumidifier_power = OpenStudio::Model::EnergyManagementSystemSensor.new(model, 'Zone Dehumidifier Electric Power')
+    dehumidifier_power.setName("#{zone_hvac.name} power htg")
+    dehumidifier_power.setKeyName(zone_hvac.name.to_s)
+
+    # actuator
+    dehumidifier_load_adj_def = OpenStudio::Model::OtherEquipmentDefinition.new(model)
+    dehumidifier_load_adj_def.setName("#{zone_hvac.name} sens htg adj def")
+    dehumidifier_load_adj_def.setDesignLevel(0)
+    dehumidifier_load_adj_def.setFractionRadiant(0)
+    dehumidifier_load_adj_def.setFractionLatent(0)
+    dehumidifier_load_adj_def.setFractionLost(0)
+    dehumidifier_load_adj = OpenStudio::Model::OtherEquipment.new(dehumidifier_load_adj_def)
+    dehumidifier_load_adj.setName("#{zone_hvac.name} sens htg adj")
+    dehumidifier_load_adj.setSpace(living_space)
+    dehumidifier_load_adj.setSchedule(model.alwaysOnDiscreteSchedule)
+
+    dehumidifier_load_adj_act = OpenStudio::Model::EnergyManagementSystemActuator.new(dehumidifier_load_adj, 'OtherEquipment', 'Power Level')
+    dehumidifier_load_adj_act.setName("#{zone_hvac.name} sens htg adj act")
+
+    # EMS program
+    program = OpenStudio::Model::EnergyManagementSystemProgram.new(model)
+    program.setName("#{zone_hvac.name} load adj program")
+    program.addLine("If #{dehumidifier_sens_htg.name} > 0")
+    program.addLine("  Set #{dehumidifier_load_adj_act.name} = - (#{dehumidifier_sens_htg.name} - #{dehumidifier_power.name}) * (1 - #{fraction_served})")
```

Comment:
@yzhou601 I'm not sure I understand this equation. The dehumidifier power (read by the sensor) already accounts for the fraction of load served, correct? So is this double counting? Would it be simpler for the sensible load adjustment to equal fraction_served * m_dot_water_dehumidifier * h_fg? I'm happy to discuss.

Reply:
Good catch! Yes the power portion is already adjusted, I will try your suggestions.

Reply:
@jonwinkler Ah wait, the total sensible heating is also only accounting for the power adjusted, so it is:

    Qsensible,avg=Mdotwater,avg * hfg+(fractioned)Pdehumid,avg

The **Qsensible,avg - (fractioned)Pdehumid,avg** should yield the same results as **m * h_fg**.

And here in the equation, I'm adjusting the load by subtracting the fraction shouldn't be handled, so the (1 - fraction) is applied

Reply:
@yzhou601 Ahh, yes, you're right! This equation is correct.